objects/screen.h

```diff
@@ -30,17 +30,27 @@
 typedef struct screen_output_t screen_output_t;
 ARRAY_TYPE(screen_output_t, screen_output)
 
+/** Different flags useful to manage screens */
+typedef enum {
+    SCREEN_FLAGS_NONE        = 0x0 << 0, /*!< No flags                       */
+    SCREEN_FLAGS_VALID       = 0x1 << 0, /*!< Not pending garbage collection */
+    SCREEN_FLAGS_FAKE        = 0x1 << 1, /*!< Is created using fake_add      */
+    SCREEN_FLAGS_MANAGED_LUA = 0x1 << 2, /*!< Is managed internally by Lua   */
+    SCREEN_FLAGS_MANAGED_C   = 0x1 << 3, /*!< Is managed internally by Lua   */
```

Comment:
`by C` instead of `by Lua`, I think...?

Could you rewrite this a bit? `NONE` can just get the value `0` and the `0x` elsewhere confuse me and can be dropped.

Also... I would split this up.
- `valid` seems to be orthogonal to the other values, so just leave that as a boolean. This also gets rid of some bit-fiddling in the implementation.
- `FAKE` is basically unused (it is set once, but never checked), so can be dropped completely.
- This leaves `MANAGES_LUA` / `MANAGED_C` / `NONE`. I am not quite sure about these. I would suggest replacing this with a `bool managed_by_c`, since i do not understand the `NONE` state. `MANAGED_LUA` is only checked for in `luaA_screen_get_managed()`, so I *think* that from the C code's perspective `NONE` and `MANAGED_BY_LUA` are the same. Lua seems to access `_managed` at one place and this could be just replaced with a table that remembers which screens are managed by `awful.screen` instead?
  - So I think that `bool managed_by_c` (or something like that) should also be an okay replacement.

Reply:
> so I think that from the C code's perspective NONE and MANAGED_BY_LUA are the same.

Not quite. If you do `c:fake_remove()` on a screen created in C to map an output. Then you do `fake_create()` with the same resolution, it would have a different, unwanted, result.

Reply:
Done-ish

```c
/** Different ways to manage screens */
typedef enum {
    SCREEN_LIFECYCLE_USER =        0, /*!< Unmanaged (ei. from fake_add) */
    SCREEN_LIFECYCLE_LUA  = 0x1 << 0, /*!< Is managed internally by Lua  */
    SCREEN_LIFECYCLE_C    = 0x1 << 1, /*!< Is managed internally by Lua  */
} screen_lifecycle_t;
```